pkg/controller/grafana/grafana_reconciler.go

```diff
@@ -64,14 +64,14 @@ func (i *GrafanaReconciler) getGrafanaReadiness(state *common.ClusterState, cr *
 			Ref: state.GrafanaRoute,
 			Msg: "check route readiness",
 		})
-	} else if !openshift && cr.Spec.Ingress != nil && cr.Spec.Ingress.Enabled {
-		// On vanilla Kubernetes, check the ingress
+	}
+	if !openshift && cr.Spec.Ingress != nil && cr.Spec.Ingress.Enabled && !cr.Spec.Client.PreferService {
+		// On vanilla Kubernetes, check the ingress,only if preferService is false
```

Comment:
Should we also skip the route check on OpenShift (above if block) when `preferService` is true?

Reply:
Yes, I'll add it in now